### PR TITLE
Feature: Column alias tables for simplified querying

### PR DESCRIPTION
**Pull request by SpencerTorres**
## Purpose

Some schemas have complicated maps such as OTel `LogAttributes`. This feature allows configuring a table like this:

```sql
CREATE TABLE otel_logs_query_aliases (
	`alias` String,
	`select` String,
	`type` String
)
Engine = MergeTree
ORDER BY (alias);
```
```sql
INSERT INTO otel_logs_query_aliases (`alias`, `select`, `type`) VALUES
('productId', 'LogAttributes[\'productId\']', 'UInt64'),
('quantity', 'LogAttributes[\'quantity\']', 'String'),
('userId', 'LogAttributes[\'userId\']', 'UInt64'),
('code.filepath', 'LogAttributes[\'code.filepath\']', 'String'),
('code.function', 'LogAttributes[\'code.function\']', 'String'),
('code.lineno', 'LogAttributes[\'code.lineno\']', 'Int64'),
. . .
```

This table can then be configured in the plugin:
config query alias tables

And then be used for the target table's column suggestions:
column suggestions

With the column selection simplified, the correct SQL will be generated using the underlying `select` syntax:
generated SQL

Notes:
 - Tables without a matching alias will default to the original `DESC TABLE` for column suggestions.
 - This capability is already built-in to ClickHouse via the [`ALIAS` column type](https://clickhouse.com/docs/en/sql-reference/statements/create/table#alias), but this feature allows for easier maintenance without having to run bulk updates on your primary table (some schemas may have thousands of alias entries).

# Changes
- added configuration for alias tables
- updated query builder to use a `label`, which defaults to the field name if an alias doesn't override it.
- added tests for affected files/features
- updated changelog

### CODE REVIEW

**Review by asimpson — DISMISSED** [review]
🏆 love it, nicely done!

File: src/components/configEditor/AliasTableConfig.tsx [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+        setEntries(removeDuplicateEntries([...entries, {
+            targetDatabase: '',
+            targetTable: '',
+            aliasDatabase: '',
+            aliasTable: ''
+        }]));
+    }
+    const removeEntry = (index: number) => {
+        let nextEntries: AliasTableEntry[] = entries.slice();
+        nextEntries.splice(index, 1);
```
[the remark this answers is not in the document]
Comment (SpencerTorres) [reply src/components/configEditor/AliasTableConfig.tsx]: Updated, good idea using these new features 😎

**Comment by SpencerTorres** [comment]
The array change you suggested worked locally, I didn't get any errors when unit testing or in-editor, but the build seemed to be failing. I tried updating to a typescript version that supports that extension and it lead to a complete collapse of the plugin's package ecosystem. I think I even saw leftpad mentioned in the logs.

Anyway, let me know if you want me to try again. I want to use that new feature, but there's a lot of dependencies affected by updating TS.

**Review by bossinc — APPROVED** [review]
LGTM!

**Pull request merged**